src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3608,6 +3609,21 @@ void Creature::InitializeInteractSpellId()
         SetInteractSpellId(0);
 }
 
+void Creature::InitializeSummonInfo()
+{
+    _summonInfo = std::make_unique<SummonInfo>(this);
+}
+
+SummonInfo* Creature::GetSummonInfo() const
```

Comment:
isn't it best to use std::optional for safer handling?

`
std::optional<SummonInfo> GetSummonInfo(uint32 summonId)
`

however, what I had in mind it would be for _summonInfo to be a std::unordered_map for faster lookup by Id